client/lib/CookieStorage.ts

```diff
@@ -38,22 +37,16 @@ export default class CookieStorage {
   }
 
   public async setItem(key: string, value: string, options?: ISetCookieOptions): Promise<boolean> {
-    const coreFrame = await getCoreFrame(this);
+    const coreFrame = await this.#coreFrame;
     return coreFrame.setCookie(key, value, options);
   }
 
   public async removeItem(name: string): Promise<boolean> {
-    const coreFrame = await getCoreFrame(this);
+    const coreFrame = await this.#coreFrame;
```

Comment:
Should this.#coreFrame be called this.#coreFramePromise? It's not really a coreFrame.